tests/system-tests/rdscore/internal/rdscorecommon/egress-ip.go

```diff
@@ -0,0 +1,519 @@
+	"github.com/openshift-kni/eco-goinfra/pkg/egressip"
+	. "github.com/openshift-kni/eco-gotests/tests/system-tests/rdscore/internal/rdscoreinittools"
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/rdscore/internal/rdscoreparams"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	proberRBACRole   = "system:openshift:scc:privileged"
+	centosIPPodLabel = "env=centos"
+)
+
+var (
+	egressIPPodLabelsMap = map[string]string{
+		strings.Split(rdscoreparams.EgressIPPodLabel, "=")[0]: strings.Split(rdscoreparams.EgressIPPodLabel, "=")[1]}
+
+	egressIPPodSelector = metav1.ListOptions{
+		LabelSelector: rdscoreparams.EgressIPPodLabel,
+	}
+)
+
+// createAgnhostRBAC creates the SCC privileged, serviceAccount and RBAC.
+func createAgnhostRBAC(
+	apiClient *clients.Settings,
+	egressIPNamespace string) (string, error) {
+	proberDeploySAName := fmt.Sprintf("rdscore-prober-sa-%s", egressIPNamespace)
+	proberDeployRBACName := fmt.Sprintf("privileged-rdscore-rbac-%s", egressIPNamespace)
+
+	var err error
+
+	glog.V(100).Infof("Adding SCC privileged to the agnhost namespace")
+
+	err = scc.AddPrivilegedSCCtoDefaultSA(egressIPNamespace)
+	if err != nil {
+		return "", fmt.Errorf("failed to add SCC privileged to the agnhost namespace %s: %w",
+			egressIPNamespace, err)
+	}
+
+	glog.V(100).Infof("Removing ServiceAccount")
+
+	err = apiobjectshelper.DeleteServiceAccount(apiClient, proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to remove serviceAccount %q from egressIPNamespace %q",
+			proberDeploySAName, egressIPNamespace)
+	}
+
+	glog.V(100).Infof("Creating ServiceAccount")
+
+	err = apiobjectshelper.CreateServiceAccount(apiClient, proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to create serviceAccount %q in egressIPNamespace %q",
+			proberDeploySAName, egressIPNamespace)
+	}
+
+	glog.V(100).Infof("Removing Cluster RBAC")
+
+	err = apiobjectshelper.DeleteClusterRBAC(apiClient, proberDeployRBACName)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to delete prober RBAC %q", proberDeployRBACName)
+	}
+
+	glog.V(100).Infof("Creating Cluster RBAC")
+
+	err = apiobjectshelper.CreateClusterRBAC(apiClient, proberDeployRBACName, proberRBACRole,
+		proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to create prober RBAC %q in egressIPNamespace %s",
+			proberDeployRBACName, egressIPNamespace)
+	}
+
+	return proberDeploySAName, nil
+}
+
+// cleanUpDeployment removes all qe deployment leftovers from the cluster according to the label.
+func cleanUpDeployment(
+	apiClient *clients.Settings,
+	nsname,
+	scheduleOnHost string) error {
+	nameSuffix := strings.Split(scheduleOnHost, ".")[0]
+	agnhostDeploymentName := fmt.Sprintf("%s-deployment-%s", nsname, nameSuffix)
+	centosDeploymentName := fmt.Sprintf("%s-centos-deployment-%s", nsname, nameSuffix)
```

Comment:
nit: it could be RHEL or Fedora so I suggest dropping _centos_ part from the name or making it more generic
```suggestion
	centosDeploymentName := fmt.Sprintf("%s-deployment-%s", nsname, nameSuffix)
```